File: ports/go/src/gui/painting/bcspainter.go
package painting

import (
	"bcs/core/tools"
)

// BcsPainter abstracts native hardware-accelerated and software drawing ops
type BcsPainter struct {
	pen   *BcsPen
	brush *BcsBrush
}

func NewBcsPainter() *BcsPainter {
	return &BcsPainter{}
}

func (p *BcsPainter) SetPen(pen *BcsPen) {
	p.pen = pen
}

func (p *BcsPainter) SetBrush(brush *BcsBrush) {
	p.brush = brush
}

func (p *BcsPainter) DrawRect(rect *tools.BcsRect) {
	// Dispatch native draw rect
}

func (p *BcsPainter) DrawLine(p1, p2 *tools.BcsPoint) {
	// Dispatch native draw line
}

File: ports/rust/src/gui/painting/bcspainter.rs
use super::bcspen::{BcsBrush, BcsPen};
use bcs_core::tools::bcsgeometry::{BcsPoint, BcsRect};

// BcsPainter abstracts native hardware-accelerated and software drawing ops
pub struct BcsPainter {
    pub pen: Option<BcsPen>,
    pub brush: Option<BcsBrush>,
}

impl BcsPainter {
    pub fn new() -> Self {
        BcsPainter {
            pen: None,
            brush: None,
        }
    }

    pub fn set_pen(&mut self, pen: BcsPen) {
        self.pen = Some(pen);
    }

    pub fn set_brush(&mut self, brush: BcsBrush) {
        self.brush = Some(brush);
    }

    pub fn draw_rect(&self, _rect: &BcsRect) {
        // Dispatch native draw rect
    }

    pub fn draw_line(&self, _p1: &BcsPoint, _p2: &BcsPoint) {
        // Dispatch native draw line
    }
}
